### DISCUSSION TITLE
Parsing error: Unexpected token import - when using import.meta.url

### DISCUSSION THREAD

**Discussion by viveleroi** [eslint/eslint#16037] [General] [answered]
eslint 8.18.0, node 14

I need to get absolute paths to a file that will be installed in a global npm package, and so far the only way to do that (while also using ESM code) is the following code. However, eslint throws a parsing error and even ignoring the next line fails to bypass this.

```
// eslint-disable-next-line
const dirname = fileURLToPath(new URL('.', import.meta.url));
```

It throws an error `Parsing error: Unexpected token import`. I used import everywhere in my code to actually import stuff, so it's only failing when it sees import used in this context.

**Comment by nzakas** [comment, marked as answer]
Assuming you are using the builtin parser, you need to have ecmaVersion set to “latest” and sourceType set to “module”.

**Reply by viveleroi** [reply]
I do. Those are for `import x from 'x';` statements

**Reply by viveleroi** [reply]
Nevermind I was wrong, I had ecmaVersion: 8. latest solves this

**Reply by jogendhra** [reply]
> Nevermind I was wrong, I had ecmaVersion: 8. latest solves this

Thanks Its work for me - Just added these lines into "eslint.config.mjs"

```
languageOptions: {
            ecmaVersion: 'latest',
            sourceType: 'module',
            parserOptions: {
                ecmaVersion: 'latest',
                sourceType: 'module',
                ecmaFeatures: { jsx: true },
            },
        },
```
